lib/drivers/oneTouchMiniDriver.js

```diff
@@ -437,20 +438,27 @@ module.exports = function (config) {
   };
 
   var prepBGData = function (progress, data) {
-    cfg.builder.setDefaults({ deviceId: 'OneTouch' + data.model + '-' + data.serialNumber });
+    //build missing data.id
+    data.id = 'OneTouch'+ data.model + data.serialNumber;
+    cfg.builder.setDefaults({ deviceId: data.id});
     var dataToPost = [];
-    for (var i = 0; i < data.bgmReadings.length; ++i) {
-      var datum = data.bgmReadings[i];
-      var smbg = cfg.builder.makeSMBG()
-        .with_value(datum.glucose)
-        .with_deviceTime(datum.displayTime)
-        .with_timezoneOffset(TZOFFSET / 60)
-        .with_time(datum.displayUtc)
-        .with_units('mg/dL')
-        .done();
-      dataToPost.push(smbg);
+    if (data.bgmReadings.length > 0) {
+      for (var i = 0; i < data.bgmReadings.length; ++i) {
+        var datum = data.bgmReadings[i];
+        var smbg = cfg.builder.makeSMBG()
+          .with_value(datum.glucose)
+          .with_deviceTime(datum.displayTime)
+          .with_timezoneOffset(TZOFFSET / 60)
```

Comment:
I know this is an oversight from before you took over the code, but timezone offset should not be a constant - it varies according to whether Daylight Savings Time is in effect. Can you refactor the code to generate the display and UTC timestamps as well as the timezone offset in the same way as is done [here](https://github.com/tidepool-org/chrome-uploader/blob/master/lib/drivers/oneTouchUltra2.js#L56) in the `buildDateTime`, particularly with the three calls to sundial's `formatDeviceTime`, `applyTimezone`, and `getOffsetFromZone` methods? Thanks!

CC @brandonarbiter @HowardLook